tests/scripts/all.sh

```diff
@@ -150,7 +150,7 @@ cleanup()
 {
     command make clean
 
-    find . -name yotta -prune -o -iname '*cmake*' -not -name CMakeLists.txt -exec rm -rf {} \+
+    find . -name yotta -prune -o -name .git -prune -o -iname '*cmake*' -not -name CMakeLists.txt -exec rm -rf {} \+
```

Comment:
Did this actually screw anything up ever? Just curious, since it sounds like a PITA to debug this.

Reply:
Ask @gilles-peskine-arm ;) (I also noticed that some of my local branches kept disappearing, but never made the connection.)

Reply:
@Patater Imagine having a git branch whose name contains `cmake`. Took me a while to understand why my head got detached. Then I [tightened this up](https://github.com/ARMmbed/mbedtls/pull/1470/commits/31b07e283321b1470c31466d10ca8a8099b73bba).